src/VisualStudio/TestUtilities/Window/EditorWindow.cs

```diff
@@ -1,30 +1,279 @@
+                }
+
+                return currentLineTextBeforeCursor;
+            }
+        }
+
+        public string CurrentLineTextAfterCursor
+        {
+            get
+            {
+                // Clear the current selected text, if any, so it is not included in the returned value
+                if (!string.IsNullOrEmpty(TextSelection.Text))
+                {
+                    TextSelection.CharRight(Extend: false, Count: 1);
+                }
+
+                // Select everything from the cursor to the end of the line
+                TextSelection.EndOfLine(Extend: true);
+
+                var currentLineTextAfterCursor = TextSelection.Text;
+
+                // Restore the cursor to its previous position
+                if (currentLineTextAfterCursor != string.Empty)
+                {
+                    TextSelection.CharLeft(false, 1);
+                }
+
+                return currentLineTextAfterCursor;
+            }
         }
 
         public string Text
         {
             get
             {
-                return _visualStudio.ExecuteOnHostProcess<string>(typeof(RemotingHelper), nameof(RemotingHelper.GetActiveTextViewContents), System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Static);
+                return _editorWindowWrapper.Contents;
             }
 
             set
             {
-                _visualStudio.ExecuteOnHostProcess<string>(typeof(RemotingHelper), nameof(RemotingHelper.SetActiveTextViewContents), System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Static, value);
+                _editorWindowWrapper.Contents = value;
             }
         }
+
+        private TextSelection TextSelection => (TextSelection)(_visualStudioInstance.Dte.ActiveDocument.Selection);
+
+        public void Activate() => _visualStudioInstance.Dte.ActiveDocument.Activate();
+
+        public void ClearTextSelection() => TextSelection?.Cancel();
+
+        public void Find(string expectedText)
+        {
+            Activate();
+            ClearTextSelection();
+
+            var dteFind = _visualStudioInstance.Dte.Find;
+
+            dteFind.Action = vsFindAction.vsFindActionFind;
+            dteFind.FindWhat = expectedText;
+            dteFind.MatchCase = true;
+            dteFind.MatchInHiddenText = true;
+            dteFind.MatchWholeWord = true;
+            dteFind.PatternSyntax = vsFindPatternSyntax.vsFindPatternSyntaxLiteral;
+            dteFind.Target = vsFindTarget.vsFindTargetCurrentDocument;
+
+            var findResult = dteFind.Execute();
+
+            if (findResult != vsFindResult.vsFindResultFound)
+            {
+                throw new Exception($"The specified text was not found. ExpectedText: '{expectedText}'; ActualText: '{Text}'");
+            }
+        }
+
+        public void PlaceCursor(string marker) => Find(marker);
+
+        public void TypeText(string text, int wordsPerMinute = 35)
```

Comment:
Based on some brief research 35 WPM is considered 'moderate' typing speed and the 'average' word is between 4 and 5 characters. So it just calculates the number of characters per second and puts a delay between each character it sends to provide more 'realistic' input.

I can remove it if we want to just slam text at the IDE at full speed...